internal/pkg/action/operator_uninstall.go

```diff
@@ -57,16 +65,41 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 		return fmt.Errorf("operator package %q not found", u.Package)
 	}
 
-	csv, err := u.getInstalledCSV(ctx, sub)
-	if err != nil {
-		return fmt.Errorf("get installed CSV %q: %v", sub.Status.InstalledCSV, err)
+	var subObj, csvObj controllerutil.Object
+	var crds []controllerutil.Object
+	if sub != nil {
```

Comment:
You're correct, this condition can be removed due to the above check. This PR is a port of https://github.com/operator-framework/operator-sdk/pull/4303 so there might be some incongruities that should be resolved. `subObj` was necessary due to some strange nil comparison behavior I experienced when writing the linked PR, and made the code a little cleaner. It might not make sense here given your comments below.